src/lib/util/cursor.c

```diff
@@ -402,6 +402,21 @@ void fr_cursor_merge(fr_cursor_t *cursor, fr_cursor_t *to_append)
 	}
 }
 
+/** Return the next item from a cursor that also satisfies an evaluation function.
+ *
+ * @param[in] cursor	to operate on
+ * @param[in] eval	evaluation function
+ * @return the next item satisfying eval, or NULL if no such item exists
+ */
+void *fr_cursor_filter_current(fr_cursor_t *cursor, bool (*eval)(void *))
```

Comment:
Can you also define a type for the callback.

Reply:
and add test cases :)

Reply:
Yes, I can and will.

Reply:
You should also add an additional `void *uctx` arg and pass it to the filter function.

Reply:
Done.